### ISSUE TITLE
Displaying live charts over 4 hours fails

### ISSUE THREAD

**Issue by BFergerson** [sourceplusplus/sourceplusplus#1041]
Example:
![live-activity-chart-fail](https://github.com/sourceplusplus/sourceplusplus/assets/3278877/935b306b-6182-40e8-ba44-c090415eca36)

Chart is generated by `LiveViewChartWindow`. Metrics can be fetched with `spp-plugin.yml` of:
```yaml
override: true
service_host: "https://tenant1.sourceplus.plus"
authorization_code: "openmma-auth"
service_name: "openmma"
```

**Comment by chess-equality** [comment]
Initial finding is that when `reservoirSize` = 720, the `xStepSize` (after) is a negative value.

Doesn't occur on `reservoirSize` = 240 and 1440.

![image](https://github.com/sourceplusplus/sourceplusplus/assets/29477940/e0fa67f9-ec4a-4cfc-be67-3d9a3cc4a076)

![image](https://github.com/sourceplusplus/sourceplusplus/assets/29477940/e3d37e6a-6c48-494e-b3c3-1de6db8e1225)

**Comment by chess-equality** [comment]
The service proxy calls return what look like normal values when transitioning from last 4 hours to last 12 hours. I am still checking the view renderer; #1071 also seems to be related to the current issue.

![service_proxy_calls_outputs](https://github.com/sourceplusplus/sourceplusplus/assets/29477940/c89a1ed5-172d-494b-b033-a38f4d878eaa)

Previous overflow error was fixed in [bugfix](https://github.com/chess-equality/interface-jetbrains/tree/bugfix/issue-1041/display-charts-over-4hours-fails) branch.

**Comment by chess-equality** [comment]
Chart display has been fixed; still fixing #1071 .

![image](https://github.com/sourceplusplus/sourceplusplus/assets/29477940/cc1e7afc-77bc-4f29-8bee-47472199ebf9)
